src/main.ts

```diff
@@ -127,19 +140,56 @@ async function run(): Promise<void> {
       );
     }
 
+    // Build environment variables.
+    const buildEnvironmentVariables = parseKVStringAndFile(
+      buildEnvVars,
+      buildEnvVarsFile,
+    );
+    const environmentVariables = parseKVStringAndFile(envVars, envVarsFile);
+
+    // Build secret environment variables.
+    const secretEnvironmentVariables: SecretEnvVar[] = [];
+    if (secretEnvVars) {
+      for (const [key, value] of Object.entries(secretEnvVars)) {
+        const secretRef = new SecretName(value);
+        secretEnvironmentVariables.push({
+          key: key,
+          projectId: secretRef.project,
+          secret: secretRef.name,
+          version: secretRef.version,
+        });
+      }
+    }
+
+    // Build secret volumes.
+    const secretVolumes: SecretVolume[] = [];
+    if (secretVols) {
+      for (const [key, value] of Object.entries(secretVols)) {
+        const mountPath = posix.dirname(key);
+        const pth = posix.basename(key);
+
+        const secretRef = new SecretName(value);
+        secretVolumes.push({
+          mountPath: mountPath,
+          projectId: secretRef.project,
+          secret: secretRef.name,
+          versions: [
```

Comment:
Does API support mounting multiple versions?

Reply:
No, which is why it's weird that it's an array. The API is... strange.

Reply:
https://cloudfunctions.googleapis.com/$discovery/rest?version=v1 for more details